src/main.c

```diff
@@ -146,6 +146,10 @@ int run_ntttcp_sender(struct ntttcp_test_endpoint *tep)
 		}
 	}
 
+	/* get TCP_INFO stats after all transfer are completed before sockets are closed */
```

Comment:
To here, all sender sockets have been closed. See it in line #243 of run_ntttcp_sender_tcp_stream().

Reply:
Closing sockets are defered to free_ntttcp_test_endpoint_and_test.

Reply:
The reason I defer the closing is to avoid handling synchronization. There could be multiple threads sending data to the receiver and getsockopt should be called before the sockets are closed. If I don't defer them, I need to add a mutex lock to protect the `ntttcp_test_endpoint_results` since it will add the current rtt to the `ntttcp_test_endpoint_results.average_rtt`. If I store the average_rtt per thread (in `ntttcp_stream_client`), there will be unneccessary memory wasted.